### ISSUE TITLE
Test 07-NoEffectReversals-Tests/01: el juego continua luego de que el oponente tiene 0 cartas en arsenal

### ISSUE THREAD

**Issue by burgosibarra** [RodrigoToroIcarte/IIC2113-2023-1#60]
En Test 07-NoEffectReversals-Tests/01, linea 1424, se observa que The Rock tiene una sola carta en el arsenal. Luego The Rock intenta jugar una carta, pero Kane la bloquea con un reversal.
Así, comienza el turno the Kane, quien hace uso de su habilidad (linea 1458), por lo que The Rock queda sin cartas en su arsenal (linea 1464). 
Pese a ser un Pin victory, el juego aun continua y Kane sigue eligiendo movimientos.

**Comment by RodrigoToroIcarte** [comment]
Hola,

Es un buen punto. Ocurre lo siguiente. Tal como dice el enunciado general del proyecto:

"_Ganas el juego si tu oponente no puede volcar más cartas desde su arsenal a su ringside mientras recibe daño. En este caso se considera un Pin Victory. También es posible ganar vía Count Out Victory cuando tu oponente no tiene más cartas en su arsenal al final de tu turno (o su turno)._"

En el caso que mencionas, si Kane decide terminar inmediatamente su turno gana por _Count Out Victory_. Pero igual puede decidir jugar una carta para ganar vía _Pin Victory_.[^1]

[^1]: Que vale más puntos en un torneo :)

**Comment by burgosibarra** [comment]
Pero eso me genera conflicto con el test 05-SuperstarAbilities/11 (desde la linea 815)
mismo comentario para 04-NoEffects/04, 07-NoEffectReversals-Tests/10
![image](https://github.com/RodrigoToroIcarte/IIC2113-2023-1/assets/53795389/85650d72-752a-4036-9a85-198cc933a082)

**Comment by RodrigoToroIcarte** [comment]
Sí, igual estaba pensando en eso. El juego técnicamente debería continuar en ese caso también. Entonces no me queda claro qué sea lo mejor:

1. Dejar los test cases cómo están.
2. Cambiar los test cases para que la habilidad de Kane termine el juego.
3. Cambiar los test cases para que el juego continúe si el daño es completado.

¿Alguna opinión?

**Comment by burgosibarra** [comment]
Quizas sea más sencillo "cortar" los test cases para que comparen hasta la penúltima jugada (no alteraría ni los códigos de los alumnos ni el suyo)
O bien, la opción 3 me parece más adecuada, ya que define una política única para todo el juego

**Comment by RodrigoToroIcarte** [comment]
Acabo de subir una nueva versión del código base para la E3. Ahora el juego continúa si el daño de la carta fue completado. Luego de eso el jugador decide terminar su turno. ¿Podrías probarlo y ver si te funciona bien?

**Comment by burgosibarra** [comment]
Me funciona super, muchas gracias!
